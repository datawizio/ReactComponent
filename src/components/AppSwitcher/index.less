@import "../../styles/variables";

.app-switcher-icon {
  color: @app-switcher-title-color;
  width: 24px;
  height: 24px;
  fill: currentColor;
}

.app-switcher-link {
  display: flex;

  &.dw-dark {
    .app-switcher-icon {
      color: @icon-color-dark-theme;
    }
  }
}

.app-switcher-container.ant-dropdown-menu {
  width: 400px;
  background-color: @body-background;
  box-shadow: @box-shadow-base;
  border-radius: 2px;
  padding: 16px;
  margin-right: 20px;
  row-gap: 8px !important;
  display: flex;
  flex-direction: column;
  border: 1px solid @border-color-base;

  @media screen and (max-width: @screen-sm) {
    width: 320px;
  }

  .other-apps-title {
    color: @text-color;
    font-size: 14px;
    font-weight: 600;
  }

  .ant-col {
    display: flex;
    align-items: center;
    gap: 8px;
    color: @text-color;
    height: auto;
    min-height: 50px;
    padding: 8px 10px;
    cursor: pointer;
    border: 1px solid @border-color-base;
    border-radius: 2px;

    .logo {
      height: 30px;
      display: flex;
      align-items: center;

      img {
        width: 30px;
      }
    }

    .text {
      height: auto;
      min-height: 30px;
      display: flex;
      flex-direction: column;
      justify-content: center;
      gap: 4px;

      & > .ant-typography {
        margin-bottom: 0;
        font-size: 12px;
        line-height: normal;
        color: fade(@text-color, 65%);
        font-weight: 400;

        &:first-child {
          color: @text-color;
          font-weight: 600 !important;
          font-size: 14px;
        }
      }
    }

    &:hover {
      background: @app-switcher-bg-color;
    }
  }
}
