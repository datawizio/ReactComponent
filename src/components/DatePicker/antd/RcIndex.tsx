import Picker from "./RcPicker";
import { PickerProps } from "rc-picker/es/Picker";
import PickerPanel, { PickerPanelProps } from "rc-picker/es/PickerPanel";
import RangePicker from "./RcRangePicker";
import { RangePickerProps } from "rc-picker/es/RangePicker";

export {
  PickerPanel,
  RangePicker,
  PickerProps,
  PickerPanelProps,
  RangePickerProps
};

export default Picker;
