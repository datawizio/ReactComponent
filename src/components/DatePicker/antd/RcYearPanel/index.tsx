import * as React from "react";
import YearHeader from "rc-picker/lib/panels/YearPanel/YearHeader";
import YearBody, { YEAR_COL_COUNT } from "./YearBody";
import type { PanelSharedProps, PanelMode } from "rc-picker/lib/interface";
import { createKeyDownHandler } from "rc-picker/lib/utils/uiUtil";
import { calendarInfo } from "../../../../utils/calendar";

export type YearPanelProps<DateType> = {
  sourceMode: PanelMode;
} & PanelSharedProps<DateType>;

export const YEAR_DECADE_COUNT = 10;

function YearPanel<DateType>(props: YearPanelProps<DateType>) {
  const {
    prefixCls,
    operationRef,
    onViewDateChange,
    generateConfig,
    value,
    viewDate,
    sourceMode,
    onSelect,
    onPanelChange
  } = props;

  let panelPrefixCls = `${prefixCls}-year-panel`;
  //@ts-ignore
  if (generateConfig.type === "fiscal" || calendarInfo.startMonth !== 0) {
    panelPrefixCls += ` ${prefixCls}-year-panel-fiscal`;
  }

  // ======================= Keyboard =======================
  operationRef.current = {
    onKeyDown: event =>
      createKeyDownHandler(event, {
        onLeftRight: diff => {
          onSelect(generateConfig.addYear(value || viewDate, diff), "key");
        },
        onCtrlLeftRight: diff => {
          onSelect(
            generateConfig.addYear(value || viewDate, diff * YEAR_DECADE_COUNT),
            "key"
          );
        },
        onUpDown: diff => {
          onSelect(
            generateConfig.addYear(value || viewDate, diff * YEAR_COL_COUNT),
            "key"
          );
        },
        onEnter: () => {
          onPanelChange(
            sourceMode === "date" ? "date" : "month",
            value || viewDate
          );
        }
      })
  };

  // ==================== View Operation ====================
  const onDecadeChange = (diff: number) => {
    const newDate = generateConfig.addYear(viewDate, diff * 10);
    onViewDateChange(newDate);
    onPanelChange(null, newDate);
  };

  return (
    <div className={panelPrefixCls}>
      <YearHeader
        {...props}
        prefixCls={prefixCls}
        onPrevDecade={() => {
          onDecadeChange(-1);
        }}
        onNextDecade={() => {
          onDecadeChange(1);
        }}
        onDecadeClick={() => {
          onPanelChange("decade", viewDate);
        }}
      />
      <YearBody
        {...props}
        prefixCls={prefixCls}
        onSelect={date => {
          onPanelChange(sourceMode === "date" ? "date" : "month", date);
          onSelect(date, "mouse");
        }}
      />
    </div>
  );
}

export default YearPanel;
