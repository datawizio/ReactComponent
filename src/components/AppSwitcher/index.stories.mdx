import { Meta, Story, Preview, ArgsTable } from "@storybook/addon-docs/blocks";
import AppSwitcher from "./index";

<Meta title="Components/AppSwitcher" component={AppSwitcher} />

# AppSwitcher

AppSwitcher is a component that displays a dropdown menu of available applications for navigation. 
It supports light and dark themes.

export const applications = [
  {
    "app_id": 30,
    "name": "PlanoHero",
    "icon": "/app-icons/ph-dark.png",
    "dark_icon": "/app-icons/ph-light.png",
    "host": "https://planohero.datawiz.io",
    "path": "/c/:client_id",
    "description": "BENTO_PH_DESCRIPTION",
    "bento_menu_description": "BENTO_PH_DESCRIPTION",
    "is_main": false
  },
  {
    "app_id": 22,
    "name": "Admin Panel",
    "icon": "/app-icons/ap-dark.png",
    "dark_icon": "/app-icons/ap-light.png",
    "host": "https://admin-panel.datawiz.io",
    "path": "/c/:client_id",
    "description": "BENTO_AP_DESCRIPTION",
    "bento_menu_description": "BENTO_AP_DESCRIPTION",
    "is_main": false
  },
  {
    "app_id": 32,
    "name": "Data Console",
    "icon": "/app-icons/dc-dark.png",
    "dark_icon": "/app-icons/dc-light.png",
    "host": "https://data-console.datawiz.io",
    "path": "/c/:client_id",
    "description": "BENTO_DC_DESCRIPTION",
    "bento_menu_description": "BENTO_DC_DESCRIPTION",
    "is_main": false
  },
  {
    "app_id": 29,
    "name": "BES",
    "icon": "/app-icons/bes-dark.png",
    "dark_icon": "/app-icons/bes-light.png", 
    "host": "https://bes.datawiz.io",
    "path": null,
    "description": "BENTO_MAIN_DESCRIPTION",
    "bento_menu_description": "BENTO_MAIN_DESCRIPTION",
    "is_main": true
  },
  {
    "app_id": 51,
    "name": "Connector",
    "icon": "/app-icons/connector-light.png",
    "dark_icon": "/app-icons/connector-dark.png",
    "host": "https://connector.datawiz.io",
    "path": "/c/:client_id",
    "description": "UPLOAD_DATA",
    "bento_menu_description": "UPLOAD_DATA",
    "is_main": false
  }
];

<Preview>
  <Story name="Basic">
    <AppSwitcher apps={applications} client="123" theme="light" />
  </Story>
</Preview>

## Props

<ArgsTable of={AppSwitcher} />