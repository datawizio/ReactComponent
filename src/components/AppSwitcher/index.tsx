import React, { useContext, useMemo } from "react";
import { Col, Dropdown, Row, Typography } from "antd";
import Icon from "@ant-design/icons";

import "./index.less";
import ConfigContext from "../ConfigProvider/context";
import clsx from "clsx";

export interface IApplication {
  app_id: number | string;
  name: string;
  description: string;
  bento_menu_description: string;
  is_main: boolean;

  logo?: string;
  host: string;
  path: string | null;
  icon?: string;
  dark_icon?: string;
  allowed?: boolean;
  clients?: { id: number; name: string }[];
  showButton?: boolean;
}

const App: React.FC<
  IApplication & { client: number; onAppClick?: (app: string) => void }
> = ({
  client,
  name,
  bento_menu_description,
  host,
  icon,
  dark_icon,
  path,
  onAppClick
}) => {
  const handleClick = () => {
    const url = `${host}${path ? path : ""}`;
    window.open(url.replace(":client_id", client.toString()), "_blank");
    onAppClick && onAppClick(name);
  };

  return (
    <Col span={24} onClick={handleClick} flex="1">
      <div className="logo">
        <img src={window.theme === "dark" ? dark_icon : icon} alt={name} />
      </div>

      <div className="text">
        <Typography.Paragraph>{name}</Typography.Paragraph>
        <Typography.Paragraph>{bento_menu_description}</Typography.Paragraph>
      </div>
    </Col>
  );
};

const menu = (
  apps: IApplication[],
  client: number,
  onAppClick?: (app: string) => void
) => {
  const mainApps = apps.filter(item => item.is_main);
  const otherApps = apps.filter(item => !item.is_main);

  return (
    <Row className="app-switcher-container">
      {mainApps.map(app => (
        <App
          {...app}
          client={client}
          key={app.app_id}
          onAppClick={onAppClick}
        />
      ))}

      <span className="other-apps-title">Перейти до</span>

      {otherApps.map(app => (
        <App
          {...app}
          client={client}
          key={app.app_id}
          onAppClick={onAppClick}
        />
      ))}
    </Row>
  );
};

const AppSwitcherSvg = () => (
  <svg className="app-switcher-icon" focusable="false" viewBox="0 0 24 24">
    <path d="M6,8c1.1,0 2,-0.9 2,-2s-0.9,-2 -2,-2 -2,0.9 -2,2 0.9,2 2,2zM12,20c1.1,0 2,-0.9 2,-2s-0.9,-2 -2,-2 -2,0.9 -2,2 0.9,2 2,2zM6,20c1.1,0 2,-0.9 2,-2s-0.9,-2 -2,-2 -2,0.9 -2,2 0.9,2 2,2zM6,14c1.1,0 2,-0.9 2,-2s-0.9,-2 -2,-2 -2,0.9 -2,2 0.9,2 2,2zM12,14c1.1,0 2,-0.9 2,-2s-0.9,-2 -2,-2 -2,0.9 -2,2 0.9,2 2,2zM16,6c0,1.1 0.9,2 2,2s2,-0.9 2,-2 -0.9,-2 -2,-2 -2,0.9 -2,2zM12,8c1.1,0 2,-0.9 2,-2s-0.9,-2 -2,-2 -2,0.9 -2,2 0.9,2 2,2zM18,14c1.1,0 2,-0.9 2,-2s-0.9,-2 -2,-2 -2,0.9 -2,2 0.9,2 2,2zM18,20c1.1,0 2,-0.9 2,-2s-0.9,-2 -2,-2 -2,0.9 -2,2 0.9,2 2,2z"></path>
  </svg>
);

const AppSwitcherIcon = () => <Icon component={AppSwitcherSvg} />;

export interface IAppSwitcher {
  apps: IApplication[];
  client: number;
  theme?: "dark" | "light";
  onAppClick?: (app: string) => void;
}

const AppSwitcher: React.FC<IAppSwitcher> = ({
  apps,
  client,
  theme,
  onAppClick
}) => {
  const { translate } = useContext(ConfigContext);

  const overlay = useMemo(() => {
    return menu(apps, client, onAppClick);
  }, [apps, client, onAppClick]);

  const className = clsx({
    "app-switcher-link": true,
    "dw-dark": theme === "dark"
  });

  return (
    <>
      <Dropdown
        overlay={overlay}
        trigger={["click"]}
        placement="bottomCenter"
        overlayClassName="AppSwitcher__dropdown"
      >
        <a
          href="#1"
          className={className}
          onClick={e => e.preventDefault()}
          title={translate("CHANGE_APP_BTN_TITLE")}
        >
          <AppSwitcherIcon />
        </a>
      </Dropdown>
      <div className="divider"></div>
    </>
  );
};

export default AppSwitcher;
