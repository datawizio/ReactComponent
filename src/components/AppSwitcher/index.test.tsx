import "jsdom-global/register";
import React from "react";
import { mount } from "enzyme";
import { Dropdown } from "antd";
import AppSwitcher from "./index";
import ConfigContext from "../ConfigProvider/context";

type AppSwitcherProps = React.ComponentProps<typeof AppSwitcher>;

const mockTranslate = (key: string) => `${key}_TRANSLATED`;

const mockProps: AppSwitcherProps = {
  apps: [
    {
      app_id: 30,
      name: "PlanoHero",
      icon: "/app-icons/ph-dark.png",
      dark_icon: "/app-icons/ph-light.png",
      host: "https://planohero.datawiz.io",
      path: "/c/:client_id",
      description: "BENTO_PH_DESCRIPTION",
      bento_menu_description: "BENTO_PH_DESCRIPTION",
      is_main: false
    },
    {
      app_id: 22,
      name: "Admin Panel",
      icon: "/app-icons/ap-dark.png",
      dark_icon: "/app-icons/ap-light.png",
      host: "https://admin-panel.datawiz.io",
      path: "/c/:client_id",
      description: "BENTO_AP_DESCRIPTION",
      bento_menu_description: "BENTO_AP_DESCRIPTION",
      is_main: false
    },
    {
      app_id: 32,
      name: "Data Console",
      icon: "/app-icons/dc-dark.png",
      dark_icon: "/app-icons/dc-light.png",
      host: "https://data-console.datawiz.io",
      path: "/c/:client_id",
      description: "BENTO_DC_DESCRIPTION",
      bento_menu_description: "BENTO_DC_DESCRIPTION",
      is_main: false
    },
    {
      app_id: 29,
      name: "BES",
      icon: "/app-icons/bes-dark.png",
      dark_icon: "/app-icons/bes-light.png",
      host: "https://bes.datawiz.io",
      path: null,
      description: "BENTO_MAIN_DESCRIPTION",
      bento_menu_description: "BENTO_MAIN_DESCRIPTION",
      is_main: true
    },
    {
      app_id: 51,
      name: "Connector",
      icon: "/app-icons/connector-light.png",
      dark_icon: "/app-icons/connector-dark.png",
      host: "https://connector.datawiz.io",
      path: "/c/:client_id",
      description: "UPLOAD_DATA",
      bento_menu_description: "UPLOAD_DATA",
      is_main: false
    }
  ],
  client: 123,
  theme: "light",
  onAppClick: jest.fn()
};

const setUp = (props?: AppSwitcherProps) =>
  mount(
    <ConfigContext.Provider value={{ translate: mockTranslate }}>
      <AppSwitcher {...props} />
    </ConfigContext.Provider>
  );

describe("AppSwitcher component", () => {
  let component: ReturnType<typeof setUp>;

  beforeEach(() => {
    component = setUp(mockProps);
  });

  it("is rendered correctly with provided props", () => {
    expect(component.find(".app-switcher-link").exists()).toBe(true);
    expect(component.find(Dropdown).exists()).toBe(true);
    expect(component).toMatchSnapshot();
  });

  it("renders the translated title for the dropdown link", () => {
    const dropdownLink = component.find(".app-switcher-link");
    expect(dropdownLink.prop("title")).toBe(
      mockTranslate("CHANGE_APP_BTN_TITLE")
    );
  });

  it("verifies that the correct app URLs are constructed", () => {
    const appNodes = component.find(".logo");
    const appUrls = mockProps.apps.map(
      app =>
        `${app.host}${
          app.path
            ? app.path.replace(":client_id", mockProps.client.toString())
            : ""
        }`
    );

    appNodes.forEach((node, index) => {
      const expectedUrl = appUrls[index];
      const onClickHandler = node.prop("onClick") as () => void;

      // Mock window.open and verify that the correct URL is passed
      onClickHandler();
      expect(window.open).toHaveBeenCalledWith(expectedUrl, "_blank");
    });
  });

  it("applies the dark theme class when theme is 'dark'", () => {
    const darkThemeComponent = setUp({ ...mockProps, theme: "dark" });
    expect(darkThemeComponent.find(".dw-dark").exists()).toBe(true);
  });

  it("handles missing or empty apps", () => {
    const emptyAppsComponent = setUp({ ...mockProps, apps: [] });
    expect(emptyAppsComponent.find(".app-switcher-container").exists()).toBe(
      false
    );
  });
});
