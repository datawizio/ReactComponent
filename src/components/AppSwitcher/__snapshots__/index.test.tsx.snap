// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppSwitcher component is rendered correctly with provided props 1`] = `
<AppSwitcher
  apps={
    Array [
      Object {
        "app_id": 30,
        "bento_menu_description": "BENTO_PH_DESCRIPTION",
        "dark_icon": "/app-icons/ph-light.png",
        "description": "BENTO_PH_DESCRIPTION",
        "host": "https://planohero.datawiz.io",
        "icon": "/app-icons/ph-dark.png",
        "is_main": false,
        "name": "PlanoHero",
        "path": "/c/:client_id",
      },
      Object {
        "app_id": 22,
        "bento_menu_description": "BENTO_AP_DESCRIPTION",
        "dark_icon": "/app-icons/ap-light.png",
        "description": "BENTO_AP_DESCRIPTION",
        "host": "https://admin-panel.datawiz.io",
        "icon": "/app-icons/ap-dark.png",
        "is_main": false,
        "name": "Admin Panel",
        "path": "/c/:client_id",
      },
      Object {
        "app_id": 32,
        "bento_menu_description": "BENTO_DC_DESCRIPTION",
        "dark_icon": "/app-icons/dc-light.png",
        "description": "BENTO_DC_DESCRIPTION",
        "host": "https://data-console.datawiz.io",
        "icon": "/app-icons/dc-dark.png",
        "is_main": false,
        "name": "Data Console",
        "path": "/c/:client_id",
      },
      Object {
        "app_id": 29,
        "bento_menu_description": "BENTO_MAIN_DESCRIPTION",
        "dark_icon": "/app-icons/bes-light.png",
        "description": "BENTO_MAIN_DESCRIPTION",
        "host": "https://bes.datawiz.io",
        "icon": "/app-icons/bes-dark.png",
        "is_main": true,
        "name": "BES",
        "path": null,
      },
      Object {
        "app_id": 51,
        "bento_menu_description": "UPLOAD_DATA",
        "dark_icon": "/app-icons/connector-dark.png",
        "description": "UPLOAD_DATA",
        "host": "https://connector.datawiz.io",
        "icon": "/app-icons/connector-light.png",
        "is_main": false,
        "name": "Connector",
        "path": "/c/:client_id",
      },
    ]
  }
  client={123}
  onAppClick={[MockFunction]}
  theme="light"
>
  <Dropdown
    mouseEnterDelay={0.15}
    mouseLeaveDelay={0.1}
    overlay={
      <ForwardRef
        className="app-switcher-container"
      >
        <App
          app_id={29}
          bento_menu_description="BENTO_MAIN_DESCRIPTION"
          client={123}
          dark_icon="/app-icons/bes-light.png"
          description="BENTO_MAIN_DESCRIPTION"
          host="https://bes.datawiz.io"
          icon="/app-icons/bes-dark.png"
          is_main={true}
          name="BES"
          onAppClick={[MockFunction]}
          path={null}
        />
        <span
          className="other-apps-title"
        >
          NAVIGATE_TO_TRANSLATED
        </span>
        <App
          app_id={30}
          bento_menu_description="BENTO_PH_DESCRIPTION"
          client={123}
          dark_icon="/app-icons/ph-light.png"
          description="BENTO_PH_DESCRIPTION"
          host="https://planohero.datawiz.io"
          icon="/app-icons/ph-dark.png"
          is_main={false}
          name="PlanoHero"
          onAppClick={[MockFunction]}
          path="/c/:client_id"
        />
        <App
          app_id={22}
          bento_menu_description="BENTO_AP_DESCRIPTION"
          client={123}
          dark_icon="/app-icons/ap-light.png"
          description="BENTO_AP_DESCRIPTION"
          host="https://admin-panel.datawiz.io"
          icon="/app-icons/ap-dark.png"
          is_main={false}
          name="Admin Panel"
          onAppClick={[MockFunction]}
          path="/c/:client_id"
        />
        <App
          app_id={32}
          bento_menu_description="BENTO_DC_DESCRIPTION"
          client={123}
          dark_icon="/app-icons/dc-light.png"
          description="BENTO_DC_DESCRIPTION"
          host="https://data-console.datawiz.io"
          icon="/app-icons/dc-dark.png"
          is_main={false}
          name="Data Console"
          onAppClick={[MockFunction]}
          path="/c/:client_id"
        />
        <App
          app_id={51}
          bento_menu_description="UPLOAD_DATA"
          client={123}
          dark_icon="/app-icons/connector-dark.png"
          description="UPLOAD_DATA"
          host="https://connector.datawiz.io"
          icon="/app-icons/connector-light.png"
          is_main={false}
          name="Connector"
          onAppClick={[MockFunction]}
          path="/c/:client_id"
        />
      </ForwardRef>
    }
    overlayClassName="AppSwitcher__dropdown"
    placement="bottomCenter"
    trigger={
      Array [
        "click",
      ]
    }
  >
    <ForwardRef(Dropdown)
      mouseEnterDelay={0.15}
      mouseLeaveDelay={0.1}
      overlay={[Function]}
      overlayClassName="AppSwitcher__dropdown"
      placement="bottomCenter"
      prefixCls="ant-dropdown"
      transitionName="slide-up"
      trigger={
        Array [
          "click",
        ]
      }
    >
      <Trigger
        action={
          Array [
            "click",
          ]
        }
        afterPopupVisibleChange={[Function]}
        autoDestroy={false}
        blurDelay={0.15}
        builtinPlacements={
          Object {
            "bottomCenter": Object {
              "offset": Array [
                0,
                4,
              ],
              "overflow": Object {
                "adjustX": 1,
                "adjustY": 1,
              },
              "points": Array [
                "tc",
                "bc",
              ],
              "targetOffset": Array [
                0,
                0,
              ],
            },
            "bottomLeft": Object {
              "offset": Array [
                0,
                4,
              ],
              "overflow": Object {
                "adjustX": 1,
                "adjustY": 1,
              },
              "points": Array [
                "tl",
                "bl",
              ],
              "targetOffset": Array [
                0,
                0,
              ],
            },
            "bottomRight": Object {
              "offset": Array [
                0,
                4,
              ],
              "overflow": Object {
                "adjustX": 1,
                "adjustY": 1,
              },
              "points": Array [
                "tr",
                "br",
              ],
              "targetOffset": Array [
                0,
                0,
              ],
            },
            "topCenter": Object {
              "offset": Array [
                0,
                -4,
              ],
              "overflow": Object {
                "adjustX": 1,
                "adjustY": 1,
              },
              "points": Array [
                "bc",
                "tc",
              ],
              "targetOffset": Array [
                0,
                0,
              ],
            },
            "topLeft": Object {
              "offset": Array [
                0,
                -4,
              ],
              "overflow": Object {
                "adjustX": 1,
                "adjustY": 1,
              },
              "points": Array [
                "bl",
                "tl",
              ],
              "targetOffset": Array [
                0,
                0,
              ],
            },
            "topRight": Object {
              "offset": Array [
                0,
                -4,
              ],
              "overflow": Object {
                "adjustX": 1,
                "adjustY": 1,
              },
              "points": Array [
                "br",
                "tr",
              ],
              "targetOffset": Array [
                0,
                0,
              ],
            },
          }
        }
        defaultPopupVisible={false}
        destroyPopupOnHide={false}
        focusDelay={0}
        getDocument={[Function]}
        getPopupClassNameFromAlign={[Function]}
        hideAction={Array []}
        mask={false}
        maskClosable={true}
        mouseEnterDelay={0.15}
        mouseLeaveDelay={0.1}
        onPopupAlign={[Function]}
        onPopupVisibleChange={[Function]}
        overlay={[Function]}
        popup={[Function]}
        popupAlign={Object {}}
        popupClassName="AppSwitcher__dropdown"
        popupPlacement="bottomCenter"
        popupStyle={Object {}}
        popupTransitionName="slide-up"
        prefixCls="ant-dropdown"
        showAction={Array []}
        stretch="minWidth"
      >
        <a
          className="ant-dropdown-trigger app-switcher-link"
          href="#1"
          key="trigger"
          onClick={[Function]}
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          title="CHANGE_APP_BTN_TITLE_TRANSLATED"
        >
          <AppSwitcherIcon>
            <AntdIcon
              component={[Function]}
            >
              <span
                className="anticon"
                role="img"
              >
                <AppSwitcherSvg
                  aria-hidden="true"
                  className=""
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  width="1em"
                >
                  <svg
                    className="app-switcher-icon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M6,8c1.1,0 2,-0.9 2,-2s-0.9,-2 -2,-2 -2,0.9 -2,2 0.9,2 2,2zM12,20c1.1,0 2,-0.9 2,-2s-0.9,-2 -2,-2 -2,0.9 -2,2 0.9,2 2,2zM6,20c1.1,0 2,-0.9 2,-2s-0.9,-2 -2,-2 -2,0.9 -2,2 0.9,2 2,2zM6,14c1.1,0 2,-0.9 2,-2s-0.9,-2 -2,-2 -2,0.9 -2,2 0.9,2 2,2zM12,14c1.1,0 2,-0.9 2,-2s-0.9,-2 -2,-2 -2,0.9 -2,2 0.9,2 2,2zM16,6c0,1.1 0.9,2 2,2s2,-0.9 2,-2 -0.9,-2 -2,-2 -2,0.9 -2,2zM12,8c1.1,0 2,-0.9 2,-2s-0.9,-2 -2,-2 -2,0.9 -2,2 0.9,2 2,2zM18,14c1.1,0 2,-0.9 2,-2s-0.9,-2 -2,-2 -2,0.9 -2,2 0.9,2 2,2zM18,20c1.1,0 2,-0.9 2,-2s-0.9,-2 -2,-2 -2,0.9 -2,2 0.9,2 2,2z"
                    />
                  </svg>
                </AppSwitcherSvg>
              </span>
            </AntdIcon>
          </AppSwitcherIcon>
        </a>
      </Trigger>
    </ForwardRef(Dropdown)>
  </Dropdown>
  <div
    className="divider"
  />
</AppSwitcher>
`;
